### ISSUE TITLE
mysql returner won't connect if ssl parameters explicitly set to None

### ISSUE THREAD

**Issue by onorua** [saltstack/salt#36695]
### Description of Issue/Question

returner configurationis:
```
cat /etc/salt/master.d/returner.conf
master_job_cache: mysql
mysql.host: 'mysqlmost'
mysql.user: 'salt'
mysql.pass: 'pass'
mysql.db: 'salt'
mysql.port: 3306
mysql.ssl_ca: None
mysql.ssl_cert: None
mysql.ssl_key: None
```
as a result, I've got these messages:

```
Sep 29 21:15:20 salt salt-master[2529]: [ERROR   ] Error in function _return:
Sep 29 21:15:20 salt salt-master[2529]: Traceback (most recent call last):
Sep 29 21:15:20 salt salt-master[2529]:   File "/usr/lib/python2.7/dist-packages/salt/master.py", line 1558, in run_func
Sep 29 21:15:20 salt salt-master[2529]:     ret = getattr(self, func)(load)
Sep 29 21:15:20 salt salt-master[2529]:   File "/usr/lib/python2.7/dist-packages/salt/master.py", line 1371, in _return
Sep 29 21:15:20 salt salt-master[2529]:     self.opts, load, event=self.event, mminion=self.mminion)
Sep 29 21:15:20 salt salt-master[2529]:   File "/usr/lib/python2.7/dist-packages/salt/utils/job.py", line 49, in store_job
Sep 29 21:15:20 salt salt-master[2529]:     mminion.returners[saveload_fstr](load['jid'], load)
Sep 29 21:15:20 salt salt-master[2529]:   File "/usr/lib/python2.7/dist-packages/salt/returners/mysql.py", line 299, in save_load
Sep 29 21:15:20 salt salt-master[2529]:     with _get_serv(commit=True) as cur:
Sep 29 21:15:20 salt salt-master[2529]:   File "/usr/lib/python2.7/contextlib.py", line 17, in __enter__
Sep 29 21:15:20 salt salt-master[2529]:     return self.gen.next()
Sep 29 21:15:20 salt salt-master[2529]:   File "/usr/lib/python2.7/dist-packages/salt/returners/mysql.py", line 236, in _get_serv
Sep 29 21:15:20 salt salt-master[2529]:     raise salt.exceptions.SaltMasterError('MySQL returner could not connect to database: {exc}'.format(exc=exc))
Sep 29 21:15:20 salt salt-master[2529]: SaltMasterError: MySQL returner could not connect to database: (2026, "SSL connection error: SSL is required but the server doesn't support it")
```

Without specified ssl parameters we've got messages such as:
```
Sep 29 21:16:33 salt salt-master[4942]: [INFO    ] Using default for mysql ssl_key
Sep 29 21:16:33 salt salt-master[4942]: [INFO    ] Using default for mysql ssl_cert
Sep 29 21:16:33 salt salt-master[4942]: [INFO    ] Using default for mysql ssl_ca
```

for each and every salt-master returner action. 

### Versions Report
```
Salt Version:
           Salt: 2016.3.3

Dependency Versions:
           cffi: 1.5.2
       cherrypy: 2.3.0
       dateutil: 2.4.2
          gitdb: 0.6.4
      gitpython: 2.0.6
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: 1.3.7
      pycparser: 2.14
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.12 (default, Jul  1 2016, 15:12:24)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: 0.2.4
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: Ubuntu 16.04 xenial
        machine: x86_64
        release: 4.4.0-36-generic
         system: Linux
        version: Ubuntu 16.04 denial
```

**Comment by Ch3LL** [comment]
@onorua nice find! and thanks for the PR! We will close this up once your PR has been reviewed and merged! thanks :)

**Comment by cachedout** [comment]
Fixed via #36697
